srtcore/channel.cpp

```diff
@@ -390,7 +390,7 @@ int CChannel::sendto(const sockaddr* addr, CPacket& packet) const
    // convert control information into network order
    // XXX USE HtoNLA!
    if (packet.isControl())
-      for (int i = 0, n = packet.getLength() / 4; i < n; ++ i)
+      for (ptrdiff_t i = 0, n = (ptrdiff_t) packet.getLength() / 4; i < n; ++i)
```

Comment:
Neither `i` nor `n` are the *result* of pointer arithmetic. The pointers are *not* of ptrdiff_t type. So why the switch to ptrdiff_t?